src/webgpu/util/texture/texelData.ts

```diff
@@ -322,170 +155,364 @@ class TexelDataRepresentationImpl implements TexelDataRepresentation {
+    'r16sint':               makeIntegerInfo(      kR, 16, { signed:  true }),
+    'r16float':              makeFloatInfo(        kR, 16),
+    'rg8unorm':              makeNormalizedInfo(  kRG,  8, { signed: false, sRGB: false }),
+    'rg8snorm':              makeNormalizedInfo(  kRG,  8, { signed:  true, sRGB: false }),
+    'rg8uint':               makeIntegerInfo(     kRG,  8, { signed: false }),
+    'rg8sint':               makeIntegerInfo(     kRG,  8, { signed:  true }),
+    'r32uint':               makeIntegerInfo(      kR, 32, { signed: false }),
+    'r32sint':               makeIntegerInfo(      kR, 32, { signed:  true }),
+    'r32float':              makeFloatInfo(        kR, 32),
+    'rg16uint':              makeIntegerInfo(     kRG, 16, { signed: false }),
+    'rg16sint':              makeIntegerInfo(     kRG, 16, { signed:  true }),
+    'rg16float':             makeFloatInfo(       kRG, 16),
+    'rgba8unorm':            makeNormalizedInfo(kRGBA,  8, { signed: false, sRGB: false }),
+    'rgba8unorm-srgb':       makeNormalizedInfo(kRGBA,  8, { signed: false, sRGB:  true }),
+    'rgba8snorm':            makeNormalizedInfo(kRGBA,  8, { signed:  true, sRGB: false }),
+    'rgba8uint':             makeIntegerInfo(   kRGBA,  8, { signed: false }),
+    'rgba8sint':             makeIntegerInfo(   kRGBA,  8, { signed:  true }),
+    'bgra8unorm':            makeNormalizedInfo(kBGRA,  8, { signed: false, sRGB: false }),
+    'bgra8unorm-srgb':       makeNormalizedInfo(kBGRA,  8, { signed: false, sRGB:  true }),
+    'rg32uint':              makeIntegerInfo(     kRG, 32, { signed: false }),
+    'rg32sint':              makeIntegerInfo(     kRG, 32, { signed:  true }),
+    'rg32float':             makeFloatInfo(       kRG, 32),
+    'rgba16uint':            makeIntegerInfo(   kRGBA, 16, { signed: false }),
+    'rgba16sint':            makeIntegerInfo(   kRGBA, 16, { signed:  true }),
+    'rgba16float':           makeFloatInfo(     kRGBA, 16),
+    'rgba32uint':            makeIntegerInfo(   kRGBA, 32, { signed: false }),
+    'rgba32sint':            makeIntegerInfo(   kRGBA, 32, { signed:  true }),
+    'rgba32float':           makeFloatInfo(     kRGBA, 32),
+  },
+  ...{
+    rgb10a2unorm: {
+      componentOrder: kRGBA,
+      componentInfo: {
+        [TexelComponent.R]: { dataType: 'unorm', bitLength: 10 },
+        [TexelComponent.G]: { dataType: 'unorm', bitLength: 10 },
+        [TexelComponent.B]: { dataType: 'unorm', bitLength: 10 },
+        [TexelComponent.A]: { dataType: 'unorm', bitLength: 2 },
+      },
+      encode: components => {
+        components = applyEach(
+          (n: number) => (floatAsNormalizedInteger(n, 10, false), n),
+          kRGB
+        )(components);
+        components = applyEach((n: number) => (floatAsNormalizedInteger(n, 2, false), n), [
+          TexelComponent.A,
+        ])(components);
+        return components;
```

Comment:
I think perhaps this could be more clearly written as a literal with `{ R: ..., G: ..., B: ..., A: ... }` and a little repetition.

Or if the floatAsNormalizedInteger results are supposed to be discarded, simply call those 4 and then return `components` unmodified?

Reply:
Done. The discarding was a bug.